src/components/partnership/partnership.service.ts

```diff
@@ -93,6 +95,12 @@ export class PartnershipService {
 
     await this.verifyRelationshipEligibility(projectId, partnerId);
 
+    const isFirstPartnership = await this.isFirstPartnership(projectId);
+    if (!isFirstPartnership && input.primary) {
+      await this.movePrimaryPartnership(false, { projectId: projectId });
+    }
```

Comment:
Could this be moved after the partnership is created in order to simplify the move query? A partnershipId could always be given then.